kubeapps.jsonnet

```diff
@@ -10,13 +13,89 @@ local kube = import "kube.libsonnet";
   kubeless: (import "kubeless.jsonnet"),
   tiller: (import "tiller.jsonnet"),
   ssecrets: (import "sealed-secrets.jsonnet"),
+  nginx: (import "ingress-nginx.jsonnet") {
+    service+: {
+      spec+: {
+        local maybe_https = if tls then [
+          {name: "https", port: 443, protocol: "TCP"},
+        ] else [],
+
+        ports: [
+          {name: "http", port: 80, protocol: "TCP"},
+        ] + maybe_https,
+      },
+    },
+  },
+
+  kubelessui: (import "kubeless-ui.jsonnet") {
+    namespace:: $.namespace,
+  },
 
-  hub: (import "kubeapps-hub.jsonnet") + {
+  hub: (import "kubeapps-hub.jsonnet") {
     namespace:: $.namespace,
     mongodb:: $.mongodb.svc,
+    ingress:: null,
+    values+: {
+      api+: {
+        service+: {type: "ClusterIP"},
+        // FIXME: api server downloads metadata/icons/etc for *every
+        // chart* *before* it starts answering /healthz
+        livenessProbe+: {initialDelaySeconds: 10*60},
+      },
+      ui+: {service+: {type: "ClusterIP"}},
+      prerender+: {service+: {type: "ClusterIP"}},
+    },
+
+    // FIXME(gus): I think these are bugs in the monocular chart
```

Comment:
is this related to the API restarting due to health failures and eventually getting to the running state? It's due to the api downloading stuff for every chart and as the chart repos grow the time for this gets longer and longer. We haven't upped the readiness/liveness time in the chart in a while, we could just set it to a high value for now.

Agree that we can fix this by making `/healthz` responsive straight away, but I also want to extract the downloading of metadata/icons/etc. to a separate job

Reply:
Here, I think the readinessProbe should not have an initialDelay.  There's very little reason to avoid sending traffic to the jobs as soon as they can respond correctly - in particular, even before the (independent) health check has passed.

Currently, the charts have an excessive _readiness_ initialDelay, presumably it was just copied from the _liveness_ initialDelay (which the api job needs, as currently implemented).  All this readiness delay does is delay availability during rollouts/bringup.

Reply:
yeah you're right, I think we can indeed remove the readiness initialDelay